pkg/server/status_test.go

```diff
@@ -668,3 +668,59 @@ func TestHotRangesPayloadMultitenant(t *testing.T) {
 		return nil
 	})
 }
+
+func TestHotRangesByNode(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	sc := log.ScopeWithoutShowLogs(t)
+	defer sc.Close(t)
+
+	ctx := context.Background()
+
+	tc := serverutils.StartCluster(t, 3, base.TestClusterArgs{
+		ReplicationMode: base.ReplicationManual,
+		ServerArgs:      base.TestServerArgs{},
+	})
+	defer tc.Stopper().Stop(ctx)
+
+	// first verify that passing no ranges, applies a fanout
+	// eg returns the same results, regardless of which node is queried.
+	testutils.SucceedsSoon(t, func() error {
+		req := &serverpb.HotRangesRequest{PageSize: 50}
+		resp1, err := tc.ApplicationLayer(0).StatusServer().(*systemStatusServer).HotRangesV2(ctx, req)
+		if err != nil {
+			return err
+		}
+		resp2, err := tc.ApplicationLayer(1).StatusServer().(*systemStatusServer).HotRangesV2(ctx, req)
+		if err != nil {
+			return err
+		}
+		if len(resp1.Ranges) == 0 {
+			return errors.New("waiting for ranges")
+		}
+
+		require.Equal(t, resp1.Ranges, resp2.Ranges)
+
+		return nil
+	})
+
+	// then check that specifying a node results in different results.
+	testutils.SucceedsSoon(t, func() error {
```

Comment:
since we're adding new functionality to query multiple nodes at the same time, could we add a test for that as well (i.e a request with `len(Nodes) > 1`

Also a test case for `NodeID` since we're still supporting it.

Might be the same test (i.e `req.NodeID="1"` and `req.Nodes = ["2"]` resulting in ranges from node 1 and 2).

Reply:
Definitely, going to add one now.